tests/test_gui.py

```diff
@@ -0,0 +1,1018 @@
+
+    # Click to create a new note
+    if app_dialog.root.captions["Click to create a new note..."].exists():
+        app_dialog.root.captions["Click to create a new note..."].get().mouseClick()
+
+    # Validate that all buttons are available
+    assert app_dialog.root.buttons[
+        "Cancel"
+    ].exists(), "Cancel buttons is not showing up"
+    assert app_dialog.root.buttons[
+        "Attach Files"
+    ].exists(), "Attach Screenshot buttons is not showing up"
+    assert app_dialog.root.buttons[
+        "Take Screenshot"
+    ].exists(), "Take Screenshot buttons is not showing up"
+    assert app_dialog.root.buttons[
+        "Create Note"
+    ].exists(), "Create Note buttons is not showing up"
+
+    # Validate that the File browser is showing up after clicking on the Files to attach button then close it
+    app_dialog.root.buttons["Attach Files"].get().mouseClick()
+    app_dialog.root.dialogs["Select files to attach."].waitExist(), 30
+    app_dialog.root.dialogs["Select files to attach."].buttons[
+        "Close"
+    ].get().mouseClick()
+
+    # Validate that all buttons are available
+    assert app_dialog.root.buttons["Cancel"].exists(), "Cancel button is not showing up"
+    assert app_dialog.root.buttons[
+        "add_button"
+    ].exists(), "Add attachments button is not showing up"
+    assert app_dialog.root.buttons[
+        "remove_button"
+    ].exists(), "Remove attachments button is not showing up"
+    assert app_dialog.root.buttons[
+        "Create Note"
+    ].exists(), "Create Note button is not showing up"
+    app_dialog.root.buttons["Cancel"].get().mouseClick()
+
+    # Take a screenshot
+    app_dialog.root.buttons["Take Screenshot"].get().mouseClick()
+    MyOGL = first(app_dialog.root)
+    width, height = MyOGL.size
+    MyOGL.mouseSlide(width * 0, height * 0)
+    MyOGL.mouseDrag(width * 1, height * 1)
+
+    # Add a note
+    app_dialog.root.textfields.typeIn("New Note")
+    app_dialog.root.buttons["Create Note"].get().mouseClick()
+    app_dialog.root.captions["Click to create a new note..."].waitExist(), 30
+
+
+def test_overlay(app_dialog):
+    # Click on the Overlay widget
+    app_dialog.open_demo_pane("Overlay")
+    assert app_dialog.root.captions["Overlay"].exists(), "Not on the Overlay widget"
+
+    # Validate "When the overlay widget is hidden, you should be able to click this button" button
+    app_dialog.root.buttons[
+        "When the overlay widget is hidden, you should be able to click this button."
+    ].get().mouseClick()
+    assert app_dialog.root.captions[
+        "*This is a label widget with an OverlayWidget parented to it. When shown,*"
+    ].exists(), "Message isn't showing up in the overlay dialog"
+
+    # Type in text dialog
+    app_dialog.root.textfields.typeIn("This is a test")
+
+    # Validate spinner button
+    app_dialog.root.buttons["start_spin()"].get().mouseClick()
+    # FIXME Need to find a way to know that spinner is showing up
```

Comment:
Have you found one? Maybe a screenshot? It's not the end of the world if we don't do it.